Tensile/KernelWriterAssembly.py

```diff
@@ -1118,24 +1118,33 @@ def initKernel(self, kernel, tPA, tPB ):
 
     # registers per element
     self.bpr = 4 # all registers are 32bit
+    # default setup
     self.bpeAB = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
-    self.bpeCexternal = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
-    self.bpeDexternal = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
+    self.bpeCexternal = int(self.bpr * kernel["ProblemType"]["DestDataType"].numRegisters())
+    self.bpeDexternal = int(self.bpr * kernel["ProblemType"]["DestDataType"].numRegisters())
+    # already covers: dgemm, cgemm, zgemm, sgemm
+    #               : hgemm  + !HPA ([H/H/H] compute = internal = f16)
+    #               : hgemm  +  HPA ([H/H/S] compute = internal = f32) -> new
+    #               : bfgemm +  HPA (compute = internal = f32)
+    #               : int8x4-gemm   (internal = i32)
+    # special cases : hgemm  +  HPA ([H/H/H] compute = f16, but internal = f32)
+    self.bpeCinternal = int(self.bpr * kernel["ProblemType"]["ComputeDataType"].numRegisters())
 
     #jgolds Need to check device for support
     if kernel["ProblemType"]["HighPrecisionAccumulate"]:
-        if kernel["ProblemType"]["DataType"].isHalf() or kernel["ProblemType"]["DataType"].isBFloat16():
-            self.bpeCinternal = int(self.bpr*1)
-            self.bpeDexternal = int(self.bpr*1) if kernel["_GlobalAccumulation"] else self.bpeDexternal
-        elif kernel["ProblemType"]["DataType"].isInt8x4():
-            # numRegisters for Int8x4 = numRegisters for float = 1
-            self.bpeCinternal = int(self.bpr* kernel["ProblemType"]["DataType"].numRegisters())
-        else:
-            print("HighPrecisionAccumulate only valid when DataType is half, Int8x4.")
-            self.bpeCinternal = int(self.bpr*kernel["ProblemType"]["DataType"].numRegisters())
-            kernel["ProblemType"]["HighPrecisionAccumulate"] = False
-    else:
-        self.bpeCinternal = int(self.bpr * kernel["ProblemType"]["DataType"].numRegisters())
+      # GSU
+      if kernel["ProblemType"]["DataType"].isHalf() or kernel["ProblemType"]["DataType"].isBFloat16():
+        self.bpeDexternal = int(self.bpr*1) if kernel["_GlobalAccumulation"] else self.bpeDexternal
```

Comment:
The accum buffer needs `bpeCinternal` bytes per element of ComputeDataType. Why does not global accum buffer follow that convention? `bpe[CD]external` refers to DestDataType, no?

I second @cgmillette that we need to streamline the naming. Though it's out of scope of this PR.

Reply:
I believe Dexternal is equal to Cinternal, since it is used for "accumulation". 
For example, (B,B,S)+HPA/(H,H,S)+HPA is accumulated as Single. 
(H,H,H)+HPA is also accumulated as Single which is Cinternal.

Reply:
Since Dexternal == Cinternal, then do we need the new bpeDExternal variable? Can we just use bpeCinternal only? To me that would make more sense for naming as it better describes the accumulation type.

Reply:
@cgmillette Sorry I guess I was wrong. Dexternal could be == either Cexternal or Cinternal depends on if using GSU. In a GSU kernel, the output is still a temp buffer storing data with Cinternal-typed. I just updated more comments about Dexternal you can have a look at it.

Reply:
The newly added comment makes sense now. Thanks